rllib/utils/replay_buffers/episode_replay_buffer.py

```diff
@@ -473,6 +478,8 @@ def _sample_episodes(
                 the extra model outputs at the `"obs"` in the batch is included (the
                 timestep at which the action is computed).
             finalize: If episodes should be finalized.
+            states: States of stateful `RLModule` that can be added to the
```

Comment:
I'm not sure how this would work. We sample (randomly) some episode chunk(s) from the buffer, no? So how would the user know, which states (at which timesteps and for which episodes) to provide??

Reply:
The thing is that we actually have cases where the buffer does not know how a `state` looks like as it does not know the module. For example: in behavior cloning the expert policy contains state only in rare cases, but if we want to train a stateful model we need to provide somehow states. My goal is to make the offline API as powerful as possible to apply it to real industry cases that will necessarily come with complex modules. 
A different approach would be to add a connector that does so. This is probably the "nicer" solution and more aligned with our design. What do yout hink @sven1977 ?